scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala

```diff
@@ -192,6 +192,13 @@ case class ScalafmtConfig(
     continuationIndent.callSite,
     continuationIndent.defnSite
   )
+
+  // Edition-specific settings below
```

Comment:
if we want to keep predefined edition flags (rather than comparing two editions for every token, as @olafurpg pointed out), it probably makes sense to put them here.

after we merge this, i will rebase #1551 as it uses editions as well.